packages/core/notifications/core-notifications-browser-internal/src/toasts/telemetry/event_types.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { type RootSchema, type EventTypeOpts } from '@kbn/analytics-client';
+
+export enum EventMetric {
+  TOAST_DISMISSED = 'global_toast_list_toast_dismissed',
+}
+
+export enum FieldType {
+  RECURRENCE_COUNT = 'toast_deduplication_count',
+  TOAST_MESSAGE = 'toast_message',
+  TOAST_MESSAGE_TYPE = 'toast_message_type',
+}
+
+const fields: Record<FieldType, RootSchema<Record<string, unknown>>> = {
+  [FieldType.TOAST_MESSAGE]: {
+    [FieldType.TOAST_MESSAGE]: {
+      type: 'keyword',
+      _meta: {
+        description: 'toast message text',
+        optional: false,
+      },
+    },
+  },
+  [FieldType.RECURRENCE_COUNT]: {
+    [FieldType.RECURRENCE_COUNT]: {
```

Comment:
Curious, why this wrapping into objects?

```
  [FieldType.RECURRENCE_COUNT]: {
    [FieldType.RECURRENCE_COUNT]: {
```

Reply:
it's makes it so that when composing the event schema the `FieldType.<propertyName>` can be used to access the schema property definition, whilst still having a named schema value, for what it's worth this is also an existing pattern I encountered in the codebase.